src/Engine/GameEvents/SwapCOEvent.java

```diff
@@ -0,0 +1,93 @@
+package Engine.GameEvents;
+
+import CommandingOfficers.Commander;
+import Engine.Army;
+import Engine.XYCoord;
+import Terrain.MapLocation;
+import Terrain.MapMaster;
+import UI.MapView;
+import UI.Art.Animation.GameAnimation;
+import Units.Unit;
+
+/**
+ * Does all the tag-type-specific swapping of ownership of things
+ */
+public class SwapCOEvent extends TurnEndEvent
+{
+  Commander swapTarget;
+
+  public SwapCOEvent(Army army, int turnNum, Commander swapTarget)
+  {
+    super(army, turnNum);
+    this.swapTarget = swapTarget;
+  }
+
+  @Override
+  public void performEvent(MapMaster gameMap)
+  {
+    int swapIndex = 0;
+    for( int i = 0; i < army.cos.length; ++i )
+    {
+      if( swapTarget == army.cos[i] )
+      {
+        swapIndex = i;
+        break;
+      }
+    }
+
+    Commander temp = army.cos[0];
+    army.cos[0] = army.cos[swapIndex];
+    army.cos[swapIndex] = temp;
+
+    switch (army.gameRules.tagMode)
+    {
+      case AWBW:
+        for( Unit u : army.getUnits() )
+        {
+          if( u.CO == swapTarget )
+            continue;
```

Comment:
Is it possible for this to be true?

Reply:
I don't think so. My hands are typing words.